SpatialGDK/Source/SpatialGDK/Public/Interop/RPCs/RPCService.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Interop/RPCs/RPCTypes.h"
+#include "SpatialView/SubView.h"
+
+class USpatialNetDriver;
+
+namespace SpatialGDK
+{
+class SPATIALGDK_API RPCService
+{
+public:
+	explicit RPCService(const FSubView& InRemoteSubView, const FSubView& InLocalAuthSubView);
+
+	void AdvanceView();
+
+	struct UpdateToSend
+	{
+		Worker_EntityId EntityId;
+		FWorkerComponentUpdate Update;
+		FSpatialGDKSpanId SpanId;
+	};
+	TArray<UpdateToSend> GetRPCsAndAcksToSend();
+	TArray<FWorkerComponentData> GetRPCComponentsOnEntityCreation(Worker_EntityId EntityId);
+
+	struct RPCQueueDescription
+	{
+		Worker_ComponentSetId Authority;
+		TSharedPtr<RPCBufferSender> Sender;
```

Comment:
Why are these shared pointers? I can't see why these would need shared ownership

Reply:
I need a pointer for the RPCService, and another one possibilty to a more derived type to be used externally.
A shared pointer is the easiest to use since then I do not have to care about lifecycle dependencies between the two places where these objects would appear.
I could make them UniquePointers on the user side, and raw pointers in the RPCService.

Reply:
I'd rather that as the service doesn't own it. It would make it clear what the responsibilities are.